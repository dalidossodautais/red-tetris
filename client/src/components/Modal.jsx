import React from "react";
import styled from "styled-components";
import Modal from "@material-ui/core/Modal";
import { makeStyles } from "@material-ui/core/styles";

const Paper = styled.div.attrs((props) => ({
  backgroundColor: props.theme.terciaryBackgroundColor,
  color: props.theme.primaryTextColor,
}))`
  border-style: solid;
  border-width: 2px;
  background-color: ${(props) => props.backgroundColor};
  color: ${(props) => props.color};
  width: 800px;
`;

function useClasses() {
  return makeStyles(() => ({
    modal: {
      alignItems: "center",
      display: "flex",
      justifyContent: "center",
    },
  }))();
}

export default function ({ children, onClose, onKeyUp, open }) {
  const classes = useClasses();
  return (
    <Modal
      className={classes.modal}
      onClose={onClose}
      onKeyUp={onKeyUp}
      open={!!open}
    >
      <Paper>{children}</Paper>
    </Modal>
  );
}
