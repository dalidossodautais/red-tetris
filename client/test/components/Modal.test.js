import React from "react";
import renderer from "react-test-renderer";
import Modal from "../../src/components/Modal";
import { shallow } from "enzyme";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

describe("Modal", () => {
  test("Standard", () => {
    const component = renderer.create(<Modal />);
    // rendering
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("Standard opened", () => {
    const component = shallow(<Modal open={true}></Modal>);
    // rendering
  });
});
