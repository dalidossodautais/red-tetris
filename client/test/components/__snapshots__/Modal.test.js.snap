// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Standard 1`] = `null`;

exports[`Modal Standard 2`] = `null`;
